juriscraper/pacer/mobile_query.py

```diff
@@ -0,0 +1,154 @@
+"""Parse the mobile_query.pl ("Mobile Query" menu) result.
+
+This provides the total number of docket entries, which is useful for alerts
+"""
+import pprint
+import sys
+
+from .docket_report import BaseDocketReport
+from .reports import BaseReport
+from ..lib.log_tools import make_default_logger
+from ..lib.utils import clean_court_object
+
+logger = make_default_logger()
+
+
+class MobileQuery(BaseDocketReport, BaseReport):
+    """Parse the mobile_query.pl ("Mobile Query" menu) result.
+
+    This provides the total number of docket entries, which is useful for alerts
+    """
+
+    PATH = "cgi-bin/mobile_query.pl"
+
+    CACHE_ATTRS = ["metadata"]
+
+    def __init__(self, court_id, pacer_session=None):
+        super(MobileQuery, self).__init__(court_id, pacer_session)
+        # Initialize the empty cache properties
+        self._clear_caches()
+        self._metadata = None
+
+    def parse(self):
+        self._clear_caches()
+        super(MobileQuery, self).parse()
+
+    @property
+    def metadata(self):
+        if self._metadata is not None:
+            return self._metadata
+
+        # The data we're after look like this (re-spacing):
+        # <a id="entriesLink" href="/cgi-bin/mobile_query.pl?search=dktEntry&caseid=186730&caseNum=4:06-cv-07294-PJH" data-transition="slide">
+        #                             Docket Entries
+        #                             <span class="ui-li-count">
+        #                                 29
+        #                             </span>
+        #                         </a>
+        # <div class="ui-title" style="padding-right: 90px; margin: .8em 0 1em 5%; font-size:14px; text-align: left">
+        # Pages: 1,
+        # Cost: 0.10
+
+        ui_title = self.tree.xpath('.//div[@class="ui-title"]')[0]
+        cost = ui_title.text_content().strip()
+        if (
+            cost
+        ):  # bad: means we were charged. Return immediately with bad news.
+            data = {u"cost": cost}
+            self._metadata = data
+            return data
+        span = self.tree.xpath('.//a[@id="entriesLink"]//span')[0]
+        entry_count = int(span.text_content().strip())
+        data = {}
+        data.update(
+            {
+                u"court_id": self.court_id,
+                u"entry_count": entry_count,
+                u"cost": cost,
+            }
+        )
```

Comment:
I'd make this a *little* more robust, and I wouldn't make it so specific to this one use case. For example, it should always finish parsing, and it should probably return a float for the cost.

Reply:
if it costs money, then that means it didn't find a single case, and the parsing would fail, there's no entriesLink in that case. I don't see the point of parsing in that case. But I will add something to parse the actual cost and return a float.